### DISCUSSION TITLE
Why are the math rendering options for HTML not implemented as extensions?

### DISCUSSION THREAD

**Discussion by asmaier** [jgm/pandoc#9675] [General]
There are several options in pandoc for configuring math rendering in HTML files: https://pandoc.org/MANUAL.html#math-rendering-in-html

Why are these implemented as pandoc options and not as extensions? What I mean is why is the command to convert to HTML with mathjax support like

    pandoc -t html5 --mathjax -s -o test.html test.md
    
instead of 

    pandoc -t html5+mathjax -s -o test.html test.md
    
Wouldn't it be more consistent to have math-rendering in HTML configured as extensions instead of options?

**Comment by jgm** [comment]
Maybe.  But this was all implemented before we had extensions.

**Comment by asmaier** [comment]
Thank you for the quick response. I completely understand that things happen for historical reasons. It also shows that extensions are a good abstraction, if they allow to reduce the number of options for pandoc. Would it be a huge effort to offer the same functionality as already implemented in https://pandoc.org/MANUAL.html#math-rendering-in-html via extensions?

**Reply by jgm** [reply]
Why would this help you?

**Reply by asmaier** [reply]
Actually I thought more that it might help to simplify pandoc (and the pandoc documentation) if the number of options is reduced. 
 
It just looks "nicer" if the output format is 

    pandoc -t html5+mathjax+<another_extension> ...

instead of

    pandoc --mathjax -t html5+<another_extension> ...

**Reply by jgm** [reply]
Maybe it would simplify things, but we'd have to support the options for a good while anyway, for backwards compatibility, so the payoff seems limited.
